### PR TITLE
Overview

### PR DESCRIPTION
**Pull request by MarkPThomas**
Merging this should fully sync all changes I have made, and de-cluttered main from files previously checked in to Git that should not have (or have been factored out). 

I am tempted to just merge it, but figure just for the Image Gallery component you should review first and merge the changes.

Friday morning I will double check that the changes are compatible with other changes in the app (should be, but just to be safe) and add any commits to this PR.

### CODE REVIEW

File: client/src/components/overview/ImageGallery.jsx [hunk 1]
```diff
@@ -1,53 +1,154 @@
...
+      selectedPhotoIndex: index
     });
   }
 
-  handleCollapsedView() {
-    this.setState({
-      isExpanded: false
-    });
-  }
+  // TODO: Finish dev
```
Comment (stevenharderjr) [inline client/src/components/overview/ImageGallery.jsx:32]: Love the pseudocode

File: client/src/components/overview/ImageGallery.jsx [hunk 2]
```diff
@@ -1,53 +1,154 @@
...
+    const model = new ImageGalleryModel({
+      photos: this.props?.photos,
+      isExpanded: this.props?.isExpanded,
+      isZoomed: this.props?.isZoomed
+    }, currentIndex);
+    const thumbnails = model.getPhotoThumbnailUrls();
+    const viewId = model.getViewId();
+    const attributes = model.getAttributes();
+    const photoUrl = model.getSelectedPhoto();
+    const handleImageClick = model.isExpanded ? this.props.onClickZoom : this.props.onClickExpand;
```
Comment (stevenharderjr) [inline client/src/components/overview/ImageGallery.jsx:91]: Cool

File: client/src/components/overview/ImageGalleryModel.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
+class ImageGalleryModel {
```
Comment (stevenharderjr) [inline client/src/components/overview/ImageGalleryModel.js:1]: Your level of comfort with these layers and layers of abstraction is astonishing to me. No doubt this is problematic, but I'm not qualified to determine whether the problem extends beyond my own lack of experience.

**Review by stevenharderjr — APPROVED** [review]

File: client/src/components/overview/ImageGalleryModel.js (hunk 3 above)
Comment (MarkPThomas) [reply client/src/components/overview/ImageGalleryModel.js:1]: Although experience helps to start out more abstract earlier (e.g. learn to code with generics C#! Especially in interfaces!), some of it also is a funny process to see unfold from refactoring. 

Seeing everything all together is easier to read in one pass, but with issues (repeated code, extra code, coupling, etc.). Looking at it all together and grouping it in a certain way, you can better see how to extract related functionality out for making easier code to test, re-use, and maintain... but at the cost of being as obvious everywhere. Although hopefully it improves comprehension at the lower levels of abstraction, like the React files :-)

**Pull request merged**